semantic_conventions.md

```diff
@@ -31,8 +31,9 @@ Span tags apply to **the entire Span**; as such, they apply to the entire timera
 | `peer.ipv4` | string | Remote IPv4 address as a `.`-separated tuple. E.g., `"127.0.0.1"` |
 | `peer.ipv6` | string | Remote IPv6 address as a string of colon-separated 4-char hex tuples. E.g., `"2001:0db8:85a3:0000:0000:8a2e:0370:7334"` |
 | `peer.port` | integer | Remote port. E.g., `80` |
-| `peer.service` | string | Remote service name (for some unspecified definition of `"service"`). E.g., `"elasticsearch"`, `"a_custom_microservice"`, `"memcache"` |
+| `peer.service` | string | Remote service name (for some unspecified definition of `"service"`). E.g., `"elasticsearch"`, `"a_custom_microservice"`, `"memcache"`. Meaning should correspond with values set in `service`. |
 | `sampling.priority` | integer | If greater than 0, a hint to the Tracer to do its best to capture the trace. If 0, a hint to the trace to not-capture the trace. If absent, the Tracer should use its default sampling mechanism. |
+| `service` | string  | Service name for a span which should override any default defined in tracer config for current and child spans (reasonably considered to be within the scope of that service). E.g., `"account-service"`, `"a_custom_microservice"`. Meaning should correspond with values set in `peer.service`. Note: intended use is for things like a service mesh which might have many services it is creating traces for. |
```

Comment:
> reasonably considered to be within the scope of that service

Is it really necessary? If say a service mesh process is creating spans on behalf of a service, why can't it set the `service` tag on all spans it creates?

Another example is a normal service that creates most of the spans for itself, but the very first span is created on behalf of the caller (for whatever reason, the caller couldn't do it).  With the above wording that first span will cause all spans in that service to be attributed to the caller.

Reply:
If it is changed so the tracer does not take responsibility for setting service on child spans, that would also be fine - although I'm guessing that was added to overcome the issue where all instrumentations would need to know the service and added it explicitly.